packages/foam-vscode/src/features/create-from-template.ts

```diff
@@ -69,7 +160,31 @@ async function createNoteFromTemplate(): Promise<void> {
     return;
   }
 
-  const defaultFileName = 'new-note.md';
+  const templateText = await workspace.fs.readFile(
+    Uri.joinPath(templatesDir, selectedTemplate)
+  );
+
+  const givenValues = new Map<string, string>();
+  const variables = findFoamVariables(templateText.toString());
+
+  // Always resolve FOAM_TITLE_SLUG if we have FOAM_TITLE
+  // We can use it to pre-populate the filepath.
+  if (
+    (variables.includes('FOAM_TITLE') || givenValues.has('FOAM_TITLE')) &&
+    !variables.includes('FOAM_TITLE_SLUG')
+  ) {
+    variables.push('FOAM_TITLE_SLUG');
+  }
+
+  const resolvedValues = await resolveFoamVariables(variables, givenValues);
+  const subbedText = substituteFoamVariables(
+    templateText.toString(),
+    resolvedValues
+  );
+  const snippet = new SnippetString(subbedText);
+
+  const defaultSlug = resolvedValues.get('FOAM_TITLE_SLUG') || 'new-note';
```

Comment:
this should use the title, not the slug

```
const defaultFilename = resolvedValues.get('FOAM_TITLE') || 'New Note'
```

Reply:
This is what I was referring to here: https://github.com/foambubble/foam/pull/569#discussion_r611087416